README.md

```diff
@@ -90,34 +99,61 @@ jobs:
-      - name: create an ssh key just for this run
-        run: |
-          mkdir -p ~/.ssh/
-          ssh-keygen -b 2048 -t rsa -f ~/.ssh/test-runner -c "test-key-${{ github.run_id }}" -q -N ""
-      - name: install CLI, login and add ssh key
-        uses: hostwithquantum/setup-runway@v0.2.1
-        with:
-          username: ${{ secrets.RUNWAY_USERNAME }}
-          password: ${{ secrets.RUNWAY_PASSWORD }}
-          public-key: ~/.ssh/test-runner.pub
-          private-key: ~/.ssh/test-runner
-          add-key: true
-          setup-ssh: true
-      - name: create app on runway (with a **unique name**)
-        run: runway -y app create -a my-cool-app-${{ github.run_id }}
-      - name: deploy your app to runway
-        run: runway -y app deploy
-      # this is where your tests run!
-      - name: run your e2e tests here
-        run: curl https://my-cool-app-${{ github.run_id }}.pqapp.dev/
-      # then hopefully you are done :)
-      - name: cleanup app
-        if: always()
-        run : runway -y app rm -a my-cool-app-${{ github.run_id }} || true
-      - name: cleanup key - this is brute force
-        if: always()
-        run: runway -y key rm "test-key-${{ github.run_id }}" || true
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0 # this is important!
+    - name: create the application name
+      run: echo "APP_NAME=my-app-${{ github.run_id }}" >> $GITHUB_ENV
+    - name: create an ssh key just for this run
+      run: |
+        mkdir -p ~/.ssh/
+        ssh-keygen -b 2048 -t rsa -f ~/.ssh/test-runner -c "test-key-${{ github.run_id }}" -q -N ""
+    - name: install CLI, login and add ssh key
+      uses: hostwithquantum/setup-runway@v0.2.1
+      with:
+        username: ${{ secrets.RUNWAY_USERNAME }}
+        password: ${{ secrets.RUNWAY_PASSWORD }}
+        public-key: ~/.ssh/test-runner.pub
+        private-key: ~/.ssh/test-runner
+        add-key: true
+        setup-ssh: true
+    - name: create app on runway
+      run: runway -y app create -a $APP_NAME || runway -y gitremote -a $APP_NAME
+    - name: deploy your app to runway
+      run: runway -y app deploy
+    # this is where your tests run!
+    - name: run your e2e tests here
+      run: curl https://$APP_NAME.pqapp.dev/
+    # then hopefully you are done :)
+    - name: cleanup app
+      if: always()
+      run : runway -y app rm -a $APP_NAME || true
+    - name: cleanup key - this is brute force
+      if: always()
+      run: runway -y key rm "test-key-${{ github.run_id }}" || true
+```
+
+### Preview apps
+
+In the previous example, we mentioned that keeping an app for people (humans!) to look at it, may be beneficial.
+
+The following workflow expands on that concepts and deletes an application from runway when a pull-request is closes (merged or closed without merge). This example _assumes_ that you constructed the application name like, `my-app-${{ github.event.number }}` (instead of `github.run_id`).
```

Comment:
```suggestion
The following workflow expands on those concepts and deletes an application from runway when a pull-request is closed (merged or closed without merge). This example _assumes_ that you constructed the application name like, `my-app-${{ github.event.number }}` (instead of `github.run_id`).
```

Reply:
Thänk u 😅